### PR TITLE
Added function to transfer reviews from one prof to another

### PR DESCRIPTION
**Pull request by JGulbronson**
Needs test... in fact the entire Professor model does.

### CODE REVIEW

**Comment by JGulbronson** [comment]
Added a test for the MenloCourse transfer function

**Comment by JGulbronson** [comment]
Closes #230

**Comment by JGulbronson** [comment]
Quick review? @divad12 @jlfwong @mduan @jswu 

Sorry for spamming with the tags, but I need this to merge the two profiles for the prof that emailed us.

File: models/professor.py [hunk 1]
```diff
@@ -266,6 +268,12 @@ def get_departments_taught(self):
                 group(1).upper() for uc in ucs)
         return sorted(departments_taught)
 
+    def transfer_reviews_from(self, prof):
+        mcs = user_course.MenloCourse.objects(professor_id=prof.id)
```
Comment (mduan) [inline models/professor.py]: I think you also want to do this for UserCourse (in addition to MenloCourse).

File: models/professor_test.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    def test_transfer_reviews_from(self):
+        real_prof = m.Professor(
+            id='real_prof'
+        )
+        dupe_prof = m.Professor(
+            id='dupe_prof'
+        )
+
+        DUPE_PROF_REVIEWS_START_NUM = 5
```
Comment (mduan) [inline models/professor_test.py]: I'm not sure what DUPE_PROF_REVIEWS_START_NUM means. Would NUM_DUPE_PROF_REVIEWS make more sense?

File: models/professor_test.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        for i in range(DUPE_PROF_REVIEWS_START_NUM):
+            m.MenloCourse(
+                professor_id=dupe_prof.id,
+                course_id='cs135'
+            ).save()
+
+        self.assertEqual(
+            len(m.MenloCourse.objects(professor_id=real_prof.id)),
```
Comment (mduan) [inline models/professor_test.py]: Nit: for all the places where you've used `len()`, you should be able to use `.count()` which will be more efficient.

File: models/professor.py (hunk 1 above)
Comment (mduan) [inline models/professor.py]: Also, you should remove the prof that you're transferring the reviews from so people can't accidentally add reviews for that prof anymore. Alternatively, you could have a default parameter delete_prof=False (that only if set to True will actually delete the prof), that acts as a reminder to the person using the method to consider if they need to remove the prof as well as transferring the reviews.

Comment (JGulbronson) [inline models/professor.py]: Yeah, I like the default parameter idea. Originally, I was going to call this function, then delete the duplicate prof, but I agree the default arg will act as a reminder. Good call.

File: models/professor_test.py (hunk 2 above)
Comment (JGulbronson) [inline models/professor_test.py]: Yeah, good call. I think it depends how you interpret NUM_DUPE_PROF_REVIEWS, as in

`NUM DUPE_PROF_REVIEWS` vs `NUM_DUPE_PROF REVIEWS`

**Comment by mduan** [comment]
Looks good. Feel free to merge after you verify that the failing tests aren't related (based on the travis output they don't seem to be).

**Comment by JGulbronson** [comment]
Meant to squash my commit down... my bad.

Deployed, so I can look at merging profs later tonight.

**Pull request merged**